### PR TITLE
bump version to 3.0.0, update release notes

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: RELEASE_NOTES [hunk 1]
```diff
@@ -3,8 +3,64 @@ For more detailed information, please see the git log.
...
+  - include Pillow as a proper dep for scikit-image rather than as extension, since it has deps itself (#3723)
+  - update Travis config w.r.t. changes framework config defaults and required Lmod version (#3773)
+  - don't limit parallellism to 4 in recent GCC easyconfigs (#3776, #3777, #3778)
+  - include M4 as dependency in flex 2.5.39 easyconfigs + fix consistency issues (#3782)
+  - consistently apply patch for ncurses 6.0 (#3792)
+  - eliminate dependency on mpi-mic-rt in ifort (#3793)
+  - include Autotools as build dependency in all beagle-lib and MrBayes easyconfigs (#3794)
+  - make OpenBLAS use the LAPACK version specified in the easyconfig (v0.2.18 & v0.2.19) (#3795)
+  - include original download URL for ISL source tarball in GCC easyconfigs (#3798)
+  - disable installing libiberty for binutils built with intel toolchain (#3802)
```
Comment (fgeorgatos) [inline RELEASE_NOTES]: ok! lgtm!
it took me some extra time to review this, since I went on a long excursion with the individual PRs ;-)

Comment (boegel) [inline RELEASE_NOTES]: hehe, thanks for the review @fgeorgatos!

Comment (boegel) [inline RELEASE_NOTES]: @fgeorgatos that'll have to wait until v3.0.1 or v3.1, whichever happens first, sorry

People can use `--from-pr` on it though (if they know about it ;))

**Comment by boegel** [comment]
Thanks for the review @fgeorgatos!

**Pull request merged**